rbd.c

```diff
@@ -203,6 +203,66 @@ static int tcmu_rbd_flush(struct tcmu_device *dev)
 	return rbd_flush(state->image);
 }
 
+static bool cmdset[SCSI_OP_MAX] = {
+	[COMPARE_AND_WRITE] = true,
+};
+
+static int tcmu_rbd_handle_cmd(struct tcmu_device *dev,
+			       struct tcmulib_cmd *tcmulib_cmd)
+{
+	struct tcmulib_handler *handler = tcmu_get_dev_handler(dev);
+	struct tcmur_handler *store = handler->hm_private;
+	uint8_t *cdb = tcmulib_cmd->cdb;
+	struct iovec *iovec = tcmulib_cmd->iovec;
+	size_t iov_cnt = tcmulib_cmd->iov_cnt;
+	uint8_t *sense = tcmulib_cmd->sense_buf;
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint8_t cmd;
+	ssize_t ret, l = tcmu_iovec_length(iovec, iov_cnt);
+	off_t offset = block_size * tcmu_get_lba(cdb);
+	struct iovec iov;
+	size_t half = l / 2;
+	uint32_t cmp_offset;
+
+        cmd = cdb[0];
+
+	switch (cmd) {
+	case COMPARE_AND_WRITE:
+		iov.iov_base = malloc(half);
+		if (!iov.iov_base) {
+                        tcmu_err("out of memory\n");
+                        return tcmu_set_sense_data(sense, MEDIUM_ERROR,
+                                                   ASC_READ_ERROR, NULL);
+                }
+                iov.iov_len = half;
+                ret = store->read(dev, &iov, 1, offset);
```

Comment:
nit: could be just `tcmu_rbd_read()` as control has already reached rbd handler.